service/matching/taskQueueManager.go

```diff
@@ -483,7 +491,9 @@ func (c *taskQueueManagerImpl) completeTask(task *persistencespb.AllocatedTaskIn
 	c.taskGC.Run(ackLevel)
 }
 
-func (c *taskQueueManagerImpl) renewLeaseWithRetry() (taskQueueState, error) {
+func suppressFatal(*taskQueueID) {}
+
+func (c *taskQueueManagerImpl) renewLeaseWithRetry(onFatal func(*taskQueueID)) (taskQueueState, error) {
```

Comment:
we can simplify this to a boolean flag: skipUnload. No need to have signalFatalProblem as field, just use c.engine.unloadTaskQueue here.

Reply:
I tend to dislike boolean parameters because they're often opaque at the call-site. You good with an enum here?

Reply:
I was hoping to get rid of this fatalSignalBehavior entirely, and just simply true/false to call c.engine.unloadTaskQueue here. But i can live with the enum.